### PR TITLE
New folder structure

### PR DESCRIPTION
**Pull request by marieroque**
Towards: https://github.com/giantswarm/giantswarm/issues/30769

⚠️ This PR changes all the structure of the rules files

Our idea is to have a better view of rules and their ownership.
So, we propose a new folder structure for alerting rules, recording rules and unit test rules:

image

In that PR, we move rules in specific <area>/<team> sub structure.
But, there are lots of rules related to multiple teams.
For those rules, we choose to move them in a shared folder.

We expect all teams have a look on them and update them to move them in the appropriate team.
We also noticed invalid area name in many rules, it would be nice to review area and team labels in your rules.

### Checklist

- [x] Update CHANGELOG.md
- [ ] Add [Unit tests](https://github.com/giantswarm/prometheus-rules/#testing)
- [ ] Follow [Alert structure](https://github.com/giantswarm/prometheus-rules/#how-alerts-are-structured)
- [ ] Consider [creating a dashboard](https://docs.giantswarm.io/getting-started/observability/grafana/custom-dashboards/) ([guidelines](https://intranet.giantswarm.io/docs/product/ux/guidelines/dashboards/)) (if it does not exist already) to help oncallers monitor the status of the issue.
- [ ] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).

### CODE REVIEW

**Comment by marieroque** [comment]
WIP...

File: helm/prometheus-rules/templates/alerting-rules/areas/kaas/bigmac/deployment.management-cluster.rules.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  creationTimestamp: null
+  labels:
+    {{- include "labels.common" . | nindent 4 }}
+{{- if not .Values.mimir.enabled }}
+    cluster_type: "management_cluster"
+{{- end }}
+  name: deployment.management-cluster.rules
```
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/alerting-rules/areas/kaas/bigmac/deployment.management-cluster.rules.yml:10]: I Guess you Split one file hère. You probably need to prefix thé rule name with the team name ?

File: helm/prometheus-rules/templates/alerting-rules/areas/kaas/bigmac/teleport.rules.yml [hunk 2]
```diff
@@ -15,18 +15,19 @@ spec:
...
-              "$1",
-              "secret",
-              "(.*)-teleport-join-token"
-            )
-        ) by (cluster_id, installation, provider, pipeline)
-        != sum (
-            capi_cluster_status_phase{phase="Provisioned"}
-        ) by (cluster_id, installation, provider, pipeline)
-      for: 60m
+          kube_secret_created{secret=~".*-teleport-join-token"}
```
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/alerting-rules/areas/kaas/bigmac/teleport.rules.yml:18]: That's not a valid change. Your main branch is behind

File: helm/prometheus-rules/templates/alerting-rules/areas/kaas/global/apiserver.workload-cluster.rules.yml
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/alerting-rules/areas/kaas/global/apiserver.workload-cluster.rules.yml:1]: Api server is turtles not global

File: helm/prometheus-rules/templates/alerting-rules/areas/kaas/global/bastions.rules.yml
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/alerting-rules/areas/kaas/global/bastions.rules.yml:1]: I think we use shared instead of global

File: helm/prometheus-rules/templates/alerting-rules/areas/kaas/global/disk.workload-cluster.rules.yml
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/alerting-rules/areas/kaas/global/disk.workload-cluster.rules.yml:1]: Why this change?

File: helm/prometheus-rules/templates/alerting-rules/areas/kaas/phoenix/aws.workload-cluster.rules.yml [hunk 3]
```diff
@@ -98,7 +98,7 @@ spec:
         area: kaas
         cancel_if_outside_working_hours: "true"
         severity: page
-        team: se
+        team: phoenix
```
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/alerting-rules/areas/kaas/phoenix/aws.workload-cluster.rules.yml:101]: I don't think phoenix will like it :)

**Comment by QuentinBisson** [comment]
I'm not sure it's reviewable right now. What do you think about keeping existing shared alerting rules in thé top folder and do the split afterwards in other PRs?

Also, you main branch is behind so you're commiting some old.versions of code that were fixed (liké téléport alerts)

**Comment by marieroque** [comment]
Yes I'm gonna centralise the shared rules in a shared folder.

File: helm/prometheus-rules/templates/alerting-rules/areas/kaas/global/apiserver.workload-cluster.rules.yml
Comment (marieroque) [reply helm/prometheus-rules/templates/alerting-rules/areas/kaas/global/apiserver.workload-cluster.rules.yml:1]: The rules uses `team: {{ include "providerTeam" . }}`, so it can be rocket

File: README.md [hunk 4]
```diff
@@ -212,15 +212,15 @@ This is a good example of an input series for testing a `range` query.
...
-Any other decision in the current helm chart is ignored for now (e.g. `helm/prometheus-rules/templates/alerting-rules/alertmanager-dashboard.rules.yml`)
 
 #### A word on the testing logic
 
 Here is a simplistic pseudocode view of the generate&test loop:
 ```
 for each provider from test/conf/providers:
-  for each file in helm/prometheus-rules/templates/alerting-rules:
-    generate the rule using helm template
+  for each file in test/hack/output/helm-chart/<provider>/prometheus-rules/templates/alerting-rules:
```
[the remark this answers is not in the document]
Comment (marieroque) [reply README.md:221]: They don't have to do it.
Scripts are generated all necessary files.

File: test/conf/promtool_ignore [hunk 5]
```diff
@@ -1,63 +1,52 @@
...
-templates/alerting-rules/timesync.rules.yml
-templates/alerting-rules/up.management-cluster.rules.yml
-templates/alerting-rules/vault.rules.yml
-templates/recording-rules/grafana-cloud.rules.yml
-templates/recording-rules/gs-managed-app-deployment-status.rules.yml
-templates/recording-rules/kubernetes-mixins.rules.yml
-templates/recording-rules/service-level.rules.yml
-templates/recording-rules/mimir-mixins.rules.yml
-templates/recording-rules/loki-mixins.rules.yml
+alerting-rules/areas/kaas/bigmac/dex.rules.yml
```
[the remark this answers is not in the document]
Comment (QuentinBisson) [reply test/conf/promtool_ignore:1]: I'm confused. If we have areas/kaas/ the rest should be team/bigmac to be area/kaas/team/bigmac or just kaas/bigmac but having area and not team is weird.

@giantswarm/team-atlas I also think we should have area/team/recording|alerting and not alerting/area/team. WDYT?

[the remark this answers is not in the document]
Comment (marieroque) [reply test/conf/promtool_ignore:1]: I removed `team` in the path because it was obvious for me.
Area gives more context for `kaas` and `platform` values IMHO.

For moving alerting and recording under each team, I think about it but I wanted the minimal number of changes in that PR and currently our recording rules are multi-teams.

[the remark this answers is not in the document]
Comment (QuentinBisson) [reply test/conf/promtool_ignore:1]: We need to align imo

[the remark this answers is not in the document]
Comment (QuentinBisson) [reply test/conf/promtool_ignore:1]: So yes it can be minimal changes, but I would rather we have alerting-rules at the end of the tree now and we move recording rules later

[the remark this answers is not in the document]
Comment (marieroque) [reply test/conf/promtool_ignore:1]: The path in the helm chart is:
`helm/prometheus-rules/templates/alerting-rules/areas/[kaas|platform|shared]/[team]`
`helm/prometheus-rules/templates/recording-rules`

Having `alerting-rules` under team will result to add it multiple times.

[the remark this answers is not in the document]
Comment (QuentinBisson) [reply test/conf/promtool_ignore:1]: Well we will then have `helm/prometheus-rules/templates/recording-rules/areas/[kaas|platform|shared]/[team]` later so 🤔

[the remark this answers is not in the document]
Comment (marieroque) [reply test/conf/promtool_ignore:1]: Yes it's my idea.
But for that PR as we don't have recording rules per team, I didn't change the behavior for them.

[the remark this answers is not in the document]
Comment (QuantumEnigmaa) [reply test/conf/promtool_ignore:1]: I agree with Quentin, I think it would be better to have a path like this `areas/<area_name>/<team_name>/<alerting-rules|recording-rules>` so that for a given team folder, the team has access directly to both its alerting and recording rules.

[the remark this answers is not in the document]
Comment (QuentinBisson) [reply test/conf/promtool_ignore:1]: So we we either have alerting-rules duplicated under each team, or each team duplicated under alerting-rules and recording rules. I personnally prefer the first one because all rules are in the same place. We could even add a folder for slos so we could have:

helm/prometheus-rules/templates/areas/[kaas|platform|shared]/[team]/{alerting-rules|recording-rules|slos}

[the remark this answers is not in the document]
Comment (QuentinBisson) [reply test/conf/promtool_ignore:1]: Also, I think shared should only be temporary but we can slowly get rid of it

[the remark this answers is not in the document]
Comment (marieroque) [reply test/conf/promtool_ignore:1]: I don't like the repetition of alerting-rules and recording-rules in all teams.
Maybe we can put all rules in team folder and prefix the name of the recording-rules to identify them ?

[the remark this answers is not in the document]
Comment (QuentinBisson) [reply test/conf/promtool_ignore:1]: But then you liké to have team name duplicated? I'm not sure I follow the Logic :)

[the remark this answers is not in the document]
Comment (marieroque) [reply test/conf/promtool_ignore:1]: Yes you're right.
My idea was more when recording-rules will be specific per team.
At this time, we will be able to move them in the same directory and get rid of alerting-rules/recording-rules folders.

[the remark this answers is not in the document]
Comment (QuentinBisson) [reply test/conf/promtool_ignore:1]: Let's discuss in planning :)

File: helm/prometheus-rules/templates/kaas/rocket/alerting-rules/falco.rules.yml
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/kaas/rocket/alerting-rules/falco.rules.yml:1]: That's shield right?

File: CHANGELOG.md [hunk 6]
```diff
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- /!\ Folder architecture for alerting-rules changed to fit with areas and teams.
```
Comment (QuentinBisson) [inline CHANGELOG.md:16]: I like it. What I think would make sense when you move it out of wip is that you explain what we are trying to achieve and where we want to go and ask for the other team's reviews especially about the ownership. WDYT?

File: CHANGELOG.md [hunk 7]
```diff
@@ -19,6 +19,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
```
Comment (QuentinBisson) [inline CHANGELOG.md]: Can you maybe add a bit more context?

**Comment by QuentinBisson** [comment]
They were both green so yes

**Comment by marieroque** [comment]
> They were both green so yes

No idea why tests succeeded before as the opsrecipe is really missing...

**Comment by QuentinBisson** [comment]
You're right the previous script was checking the rules the first time it so them but as those one are provider specific, they were skipped. I would definitely merge this anyway (You'll have to use you admin rights) and then we ask phoenix to provide an ops recipe
